src/include/migraphx/program.hpp

```diff
@@ -78,6 +79,7 @@ struct program
 
     std::vector<argument> eval(parameter_map params) const;
 
+    std::vector<argument> run_async(parameter_map params, execution_environment exec_env) const;
```

Comment:
For here, I would still call this `eval`, Only call it `run_async` for the API.

Reply:
Actually we dont need another method. Just add `execution_environment` parameter to `eval` and have it with default parameter:

```cpp
std::vector<argument> run_async(parameter_map params, execution_environment exec_env = execution_environment{}) const
```

Reply:
Fair I'll change this back then.